hvac/api/auth_methods/cert.py

```diff
@@ -0,0 +1,252 @@
+        :type token_max_ttl: int | str
+        :param token_policies: List of policies to encode onto generated tokens. Depending on the auth method, this list may be supplemented by user/group/other values.
+        :type token_policies: list | str
+        :param token_bound_cidrs: List of CIDR blocks; if set, specifies blocks of IP addresses which can authenticate successfully, and ties the resulting token to these blocks as well.
+        :type token_bound_cidrs: list | str
+        :param token_explicit_max_ttl: If set, will encode an explicit max TTL onto the token. This is a hard cap even if token_ttl and token_max_ttl would otherwise allow a renewal.
+        :type token_explicit_max_ttl: int | str
+        :param token_no_default_policy: If set, the default policy will not be set on generated tokens; otherwise it will be added to the policies set in token_policies.
+        :type token_no_default_policy: bool
+        :param token_num_uses: The maximum number of times a generated token may be used (within its lifetime); 0 means unlimited. If you require the token to have the ability to create child tokens, you will need to set this value to 0.
+        :type token_num_uses: int
+        :param token_period: The period, if any, to set on the token.
+        :type token_period: int | str
+        :param token_type: The type of token that should be generated. Can be service, batch, or default to use the mount's tuned default (which unless changed will be service tokens). For token store roles, there are two additional possibilities: default-service and default-batch which specify the type to return unless the client requests a different type at generation time.
+        :type token_type: str
+        :param mount_point:
+        :type mount_point:
+        """
+        try:
+            with open(certificate, 'r') as f_cert:
+                cert = f_cert.read()
+        except FileNotFoundError:
+            cert = certificate
+        params = {
+            'name': name,
+            'certificate': cert,
+            'allowed_common_names': allowed_common_names,
+            'allowed_dns_sans': allowed_dns_sans,
+            'allowed_email_sans': allowed_email_sans,
+            'allowed_uri_sans': allowed_uri_sans,
+            'allowed_organizational_units': allowed_organizational_units,
+            'required_extensions': required_extensions,
+            'display_name': display_name,
+            'token_ttl': token_ttl,
+            'token_max_ttl': token_max_ttl,
+            'token_policies': token_policies,
+            'token_bound_cidrs': token_bound_cidrs,
+            'token_explicit_max_ttl': token_explicit_max_ttl,
+            'token_no_default_policy': token_no_default_policy,
+            'token_num_uses': token_num_uses,
+            'token_period': token_period,
+            'token_type': token_type
+        }
```

Comment:
I think you'll want to wrap this in `utils.remove_nones()` to match similar methods' behavior (without that arrangement, it makes it harder for hvac users to update a single parameter without reverting the rest of them to their defaults generally):
```suggestion

        params = utils.remove_nones(
            {
                "name": name,
                "certificate": cert,
                "allowed_common_names": allowed_common_names,
                "allowed_dns_sans": allowed_dns_sans,
                "allowed_email_sans": allowed_email_sans,
                "allowed_uri_sans": allowed_uri_sans,
                "allowed_organizational_units": allowed_organizational_units,
                "required_extensions": required_extensions,
                "display_name": display_name,
                "token_ttl": token_ttl,
                "token_max_ttl": token_max_ttl,
                "token_policies": token_policies,
                "token_bound_cidrs": token_bound_cidrs,
                "token_explicit_max_ttl": token_explicit_max_ttl,
                "token_no_default_policy": token_no_default_policy,
                "token_num_uses": token_num_uses,
                "token_period": token_period,
                "token_type": token_type,
            }
        )
```

Also would need to add `from hvac import utils` to this file's imports to support that ^